// redux-openfin/system/actions/actionTypes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const makeType_1 = require("../../utils/makeType");
exports.GET_MACHINE_ID = makeType_1.makeReqType('GET_MACHINE_ID');
exports.GET_MACHINE_ID_RES = makeType_1.makeResType('GET_MACHINE_ID_RES');
exports.GET_DEVICE_USER_ID = makeType_1.makeReqType('GET_DEVICE_USER_ID');
exports.GET_DEVICE_USER_ID_RES = makeType_1.makeResType('GET_DEVICE_USER_ID_RES');
exports.GET_MONITOR_INFO = makeType_1.makeReqType('GET_MONITOR_INFO');
exports.GET_MONITOR_INFO_RES = makeType_1.makeResType('GET_MONITOR_INFO_RES');
exports.GET_VERSION = makeType_1.makeReqType('GET_VERSION');
exports.GET_VERSION_RES = makeType_1.makeResType('GET_VERSION_RES');
exports.GET_HOST_SPECS = makeType_1.makeReqType('GET_HOST_SPECS');
exports.GET_HOST_SPECS_RES = makeType_1.makeResType('GET_HOST_SPECS_RES');
exports.GET_ENVIRONMENT_VARIABLE = makeType_1.makeReqType('GET_ENVIRONMENT_VARIABLE');
exports.GET_ENVIRONMENT_VARIABLE_RES = makeType_1.makeResType('GET_ENVIRONMENT_VARIABLE_RES');
exports.CLEAR_CACHE = makeType_1.makeReqType('CLEAR_CACHE');
exports.CLEAR_CACHE_RES = makeType_1.makeResType('CLEAR_CACHE_RES');
