// redux-openfin/system/handlers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const asyncs = require("./asyncs");
//http://cdn.openfin.co/jsdocs/beta/fin.System.html#.getMachineId
exports.getMachineIdHandler = (params) => {
    const { action } = params;
    asyncs.getMachineId(action).catch((e) => {
        // eat the exception
    });
};
//http://cdn.openfin.co/jsdocs/beta/fin.System.html#.getDeviceUserId
exports.getDeviceUserIdHandler = (params) => {
    const { action } = params;
    asyncs.getDeviceUserId(action).catch((e) => {
        // eat the exception
    });
};
//http://cdn.openfin.co/jsdocs/beta/fin.System.html#.getMonitorInfo
exports.getMonitorInfoHandler = (params) => {
    const { action } = params;
    asyncs.getMonitorInfo(action).catch((e) => {
        // eat the exception
    });
};
//http://cdn.openfin.co/jsdocs/beta/fin.System.html#.getVersion
exports.getVersionHandler = (params) => {
    const { action } = params;
    asyncs.getVersion(action).catch((e) => {
        // eat the exception
    });
};
//http://cdn.openfin.co/jsdocs/beta/fin.System.html#.getHostSpecs
exports.getHostSpecsHandler = (params) => {
    const { action } = params;
    asyncs.getHostSpecs(action).catch((e) => {
        // eat the exception
    });
};
//http://cdn.openfin.co/jsdocs/beta/fin.System.html#.getEnvironmentVariable
exports.getEnvironmentVariableHandler = (params) => {
    const { action } = params;
    asyncs.getEnvironmentVariable(action).catch((e) => {
        // eat the exception
    });
};
//http://cdn.openfin.co/jsdocs/beta/fin.System.html#.clearCache
exports.clearCacheHandler = (params) => {
    const { action } = params;
    asyncs.clearCache(action).catch((e) => {
        // eat the exception
    });
};
