// redux-openfin/docking/DockingType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var GroupEventReason;
(function (GroupEventReason) {
    GroupEventReason["DISBAND"] = "disband";
    GroupEventReason["JOIN"] = "join";
    GroupEventReason["LEAVE"] = "leave";
    GroupEventReason["MERGE"] = "merge";
})(GroupEventReason = exports.GroupEventReason || (exports.GroupEventReason = {}));
var GroupEventMemberOf;
(function (GroupEventMemberOf) {
    GroupEventMemberOf["NOTHING"] = "nothing";
    GroupEventMemberOf["SOURCE"] = "source";
    GroupEventMemberOf["TARGET"] = "target";
})(GroupEventMemberOf = exports.GroupEventMemberOf || (exports.GroupEventMemberOf = {}));
var SnapDirection;
(function (SnapDirection) {
    SnapDirection[SnapDirection["RIGHT"] = 1] = "RIGHT";
    SnapDirection[SnapDirection["LEFT"] = 2] = "LEFT";
    SnapDirection[SnapDirection["BOTTOM"] = 3] = "BOTTOM";
    SnapDirection[SnapDirection["TOP"] = 4] = "TOP";
})(SnapDirection = exports.SnapDirection || (exports.SnapDirection = {}));
