// redux-openfin/event/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const makeType_1 = require("../utils/makeType");
const createFSA_1 = require("../utils/createFSA");
class EventConstant {
    constructor(prefix, name) {
        this.name = name;
        this.type = makeType_1.makeResType(prefix + name.split('-').join('_').toUpperCase());
        this.actionCreator = createFSA_1.default(this.type, (options) => options);
    }
}
exports.EventConstant = EventConstant;
const appEventStrArr = [
    'closed',
];
const windowEventStrArr = [
    // 'auth-requested',
    // 'begin-user-bounds-changing',
    'blurred',
    'bounds-changed',
    'bounds-changing',
    'close-requested',
    // 'closed',
    // 'closing',
    // 'crashed',
    // 'disabled-frame-bounds-changed',
    // 'disabled-frame-bounds-changing',
    // 'end-user-bounds-changing',
    // 'embedded',
    // 'external-process-exited',
    // 'external-process-started',
    'focused',
    // 'frame-disabled',
    // 'frame-enabled',
    'group-changed',
    'hidden',
    // 'initialized',
    'maximized',
    'minimized',
    // 'navigation-rejected',
    // 'preload-scripts-state-changed',
    // 'preload-scripts-state-changing',
    // 'reloaded',
    'restored',
    // 'show-requested',
    'shown',
];
exports.appEvents = appEventStrArr.map(one => {
    return new EventConstant('APP_EVENT::', one);
});
exports.windowEvents = windowEventStrArr.map(one => {
    return new EventConstant('WIND_EVENT::', one);
});
