// redux-openfin/notification/actions/actionTypes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const makeType_1 = require("../../utils/makeType");
exports.CREATE_NOTIFICATION = makeType_1.makeReqType('CREATE_NOTIFICATION');
exports.CREATE_NOTIFICATION_RES = makeType_1.makeResType('CREATE_NOTIFICATION_RES');
exports.NOTIFICATION_GET_CURRENT = makeType_1.makeReqType('NOTIFICATION_GET_CURRENT');
exports.NOTIFICATION_GET_CURRENT_RES = makeType_1.makeResType('NOTIFICATION_GET_CURRENT_RES');
exports.NOTIFICATION_CLOSE = makeType_1.makeReqType('NOTIFICATION_CLOSE');
exports.NOTIFICATION_CLOSE_RES = makeType_1.makeResType('NOTIFICATION_CLOSE_RES');
exports.NOTIFICATION_SEND_MSG = makeType_1.makeReqType('NOTIFICATION_SEND_MSG');
exports.NOTIFICATION_SEND_MSG_RES = makeType_1.makeResType('NOTIFICATION_SEND_MSG_RES');
exports.NOTIFICATION_SEND_MSG_TO_APP = makeType_1.makeReqType('NOTIFICATION_SEND_MSG_TO_APP');
exports.NOTIFICATION_SEND_MSG_TO_APP_RES = makeType_1.makeResType('NOTIFICATION_SEND_MSG_TO_APP_RES');
// notification events
exports.NOTIFICATION_ON_CLICK_RES = makeType_1.makeResType('NOTIFICATION_ON_CLICK_RES');
exports.NOTIFICATION_ON_CLOSE_RES = makeType_1.makeResType('NOTIFICATION_ON_CLOSE_RES');
exports.NOTIFICATION_ON_DISMISS_RES = makeType_1.makeResType('NOTIFICATION_ON_DISMISS_RES');
exports.NOTIFICATION_ON_ERROR_RES = makeType_1.makeResType('NOTIFICATION_ON_ERROR_RES');
exports.NOTIFICATION_ON_MSG_RES = makeType_1.makeResType('NOTIFICATION_ON_MSG_RES');
exports.NOTIFICATION_ON_SHOW_RES = makeType_1.makeResType('NOTIFICATION_ON_SHOW_RES');
