// redux-openfin/notification/actions/handlerActionCreator.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const createFSA_1 = require("../../utils/createFSA");
const actionTypes = require("./actionTypes");
exports.createNotificationRes = createFSA_1.default(actionTypes.CREATE_NOTIFICATION_RES, (options) => (options));
exports.getCurrentRes = createFSA_1.default(actionTypes.NOTIFICATION_GET_CURRENT_RES, (options) => (options));
exports.closeRes = createFSA_1.default(actionTypes.NOTIFICATION_CLOSE_RES, (options) => (options));
exports.sendMessageRes = createFSA_1.default(actionTypes.NOTIFICATION_SEND_MSG_RES, (options) => (options));
exports.sendMessageToApplicationRes = createFSA_1.default(actionTypes.NOTIFICATION_SEND_MSG_TO_APP_RES, (options) => (options));
// ----------------------notification events
// http://cdn.openfin.co/jsdocs/beta/fin.Notification.html#~onClick
// http://cdn.openfin.co/jsdocs/beta/fin.Notification.html#~onClose
// http://cdn.openfin.co/jsdocs/beta/fin.Notification.html#~onDismiss
// http://cdn.openfin.co/jsdocs/beta/global.html#errorCallback
// http://cdn.openfin.co/jsdocs/beta/fin.Notification.html#~onMessage
// http://cdn.openfin.co/jsdocs/beta/fin.Notification.html#~onShow
exports.notificationOnClickRes = createFSA_1.default(actionTypes.NOTIFICATION_ON_CLICK_RES, (options) => (options));
exports.notificationOnCloseRes = createFSA_1.default(actionTypes.NOTIFICATION_ON_CLOSE_RES, (options) => (options));
exports.notificationOnDismissRes = createFSA_1.default(actionTypes.NOTIFICATION_ON_DISMISS_RES, (options) => (options));
exports.notificationOnErrorRes = createFSA_1.default(actionTypes.NOTIFICATION_ON_ERROR_RES, (options) => (options));
exports.notificationOnMsgRes = createFSA_1.default(actionTypes.NOTIFICATION_ON_MSG_RES, (options) => (options));
exports.notificationOnShowRes = createFSA_1.default(actionTypes.NOTIFICATION_ON_SHOW_RES, (options) => (options));
