// redux-openfin/application/actions/actionCreators.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const createFSA_1 = require("../../utils/createFSA");
const actionTypes = require("./actionTypes");
exports.startApplication = createFSA_1.default(actionTypes.START_APPLICATION, (options) => options);
exports.getCurrent = createFSA_1.default(actionTypes.APPLICATION_GET_CURRENT, (options) => options);
exports.getWindow = createFSA_1.default(actionTypes.APPLICATION_GET_WINDOW, (options) => options);
exports.wrap = createFSA_1.default(actionTypes.APPLICATION_WRAP, (options) => options);
exports.addEventListener = createFSA_1.default(actionTypes.APPLICATION_ADD_LISTENER, (options) => options);
exports.quit = createFSA_1.default(actionTypes.APPLICATION_QUIT, (options) => options);
exports.getChildWindows = createFSA_1.default(actionTypes.APPLICATION_GET_CHILD_WINDOWS, (options) => options);
exports.getInfo = createFSA_1.default(actionTypes.APPLICATION_GET_INFO, (options) => options);
exports.getShortcuts = createFSA_1.default(actionTypes.APPLICATION_GET_SHORTCUTS, (options) => options);
exports.getTrayIconInfo = createFSA_1.default(actionTypes.APPLICATION_GET_TRAY_ICON_INFO, (options) => options);
exports.getZoomLevel = createFSA_1.default(actionTypes.APPLICATION_GET_ZOOM_LEVEL, (options) => options);
exports.isRunning = createFSA_1.default(actionTypes.APPLICATION_IS_RUNNING, (options) => options);
exports.removeListener = createFSA_1.default(actionTypes.APPLICATION_REMOVE_LISTENER, (options) => options);
exports.restart = createFSA_1.default(actionTypes.RESTART, (options) => options);
exports.scheduleRestart = createFSA_1.default(actionTypes.APPLICATION_SCHEDULE_RESTART, (options) => options);
exports.setShortcuts = createFSA_1.default(actionTypes.APPLICATION_SET_SHORTCUTS, (options) => options);
exports.setTrayIcon = createFSA_1.default(actionTypes.APPLICATION_SET_TRAY_ICON, (options) => options);
exports.setZoomLevel = createFSA_1.default(actionTypes.APPLICATION_SET_ZOOM_LEVEL, (options) => options);
exports.terminate = createFSA_1.default(actionTypes.APPLICATION_TERMINATE, (options) => options);
