// redux-openfin/application/actions/actionTypes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const makeType_1 = require("../../utils/makeType");
exports.START_APPLICATION = makeType_1.makeReqType('START_APPLICATION');
exports.START_APPLICATION_RES = makeType_1.makeResType('START_APPLICATION_RES');
exports.APPLICATION_GET_CURRENT = makeType_1.makeReqType('APPLICATION_GET_CURRENT');
exports.APPLICATION_GET_CURRENT_RES = makeType_1.makeResType('APPLICATION_GET_CURRENT_RES');
exports.APPLICATION_GET_WINDOW = makeType_1.makeReqType('APPLICATION_GET_WINDOW');
exports.APPLICATION_GET_WINDOW_RES = makeType_1.makeResType('APPLICATION_GET_WINDOW_RES');
exports.APPLICATION_WRAP = makeType_1.makeReqType('APPLICATION_WRAP');
exports.APPLICATION_WRAP_RES = makeType_1.makeResType('APPLICATION_WRAP_RES');
exports.APPLICATION_ADD_LISTENER = makeType_1.makeReqType('APPLICATION_ADD_LISTENER');
exports.APPLICATION_ADD_LISTENER_RES = makeType_1.makeResType('APPLICATION_ADD_LISTENER_RES');
exports.APPLICATION_QUIT = makeType_1.makeReqType('APPLICATION_QUIT');
exports.APPLICATION_QUIT_RES = makeType_1.makeResType('APPLICATION_QUIT_RES');
exports.APPLICATION_GET_CHILD_WINDOWS = makeType_1.makeReqType('APPLICATION_GET_CHILD_WINDOWS');
exports.APPLICATION_GET_CHILD_WINDOWS_RES = makeType_1.makeResType('APPLICATION_GET_CHILD_WINDOWS_RES');
exports.APPLICATION_GET_INFO = makeType_1.makeReqType('APPLICATION_GET_INFO');
exports.APPLICATION_GET_INFO_RES = makeType_1.makeResType('APPLICATION_GET_INFO_RES');
exports.APPLICATION_GET_SHORTCUTS = makeType_1.makeReqType('APPLICATION_GET_SHORTCUTS');
exports.APPLICATION_GET_SHORTCUTS_RES = makeType_1.makeResType('APPLICATION_GET_SHORTCUTS_RES');
exports.APPLICATION_GET_TRAY_ICON_INFO = makeType_1.makeReqType('APPLICATION_GET_TRAY_ICON_INFO');
exports.APPLICATION_GET_TRAY_ICON_INFO_RES = makeType_1.makeResType('APPLICATION_GET_TRAY_ICON_INFO_RES');
exports.APPLICATION_GET_ZOOM_LEVEL = makeType_1.makeReqType('APPLICATION_GET_ZOOM_LEVEL');
exports.APPLICATION_GET_ZOOM_LEVEL_RES = makeType_1.makeResType('APPLICATION_GET_ZOOM_LEVEL_RES');
exports.APPLICATION_IS_RUNNING = makeType_1.makeReqType('APPLICATION_IS_RUNNING');
exports.APPLICATION_IS_RUNNING_RES = makeType_1.makeResType('APPLICATION_IS_RUNNING_RES');
exports.APPLICATION_REMOVE_LISTENER = makeType_1.makeReqType('APPLICATION_REMOVE_LISTENER');
exports.APPLICATION_REMOVE_LISTENER_RES = makeType_1.makeResType('APPLICATION_REMOVE_LISTENER_RES');
exports.RESTART = makeType_1.makeReqType('APPLICATION_RESTART');
exports.RESTART_RES = makeType_1.makeResType('APPLICATION_RESTART_RES');
exports.APPLICATION_SCHEDULE_RESTART = makeType_1.makeReqType('APPLICATION_SCHEDULE_RESTART');
exports.APPLICATION_SCHEDULE_RESTART_RES = makeType_1.makeResType('APPLICATION_SCHEDULE_RESTART_RES');
exports.APPLICATION_SET_SHORTCUTS = makeType_1.makeReqType('APPLICATION_SET_SHORTCUTS');
exports.APPLICATION_SET_SHORTCUTS_RES = makeType_1.makeResType('APPLICATION_SET_SHORTCUTS_RES');
exports.APPLICATION_SET_TRAY_ICON = makeType_1.makeReqType('APPLICATION_SET_TRAY_ICON');
exports.APPLICATION_SET_TRAY_ICON_RES = makeType_1.makeResType('APPLICATION_SET_TRAY_ICON_RES');
exports.APPLICATION_SET_ZOOM_LEVEL = makeType_1.makeReqType('APPLICATION_SET_ZOOM_LEVEL');
exports.APPLICATION_SET_ZOOM_LEVEL_RES = makeType_1.makeResType('APPLICATION_SET_ZOOM_LEVEL_RES');
exports.APPLICATION_TERMINATE = makeType_1.makeReqType('APPLICATION_TERMINATE');
exports.APPLICATION_TERMINATE_RES = makeType_1.makeResType('APPLICATION_TERMINATE_RES');
