// redux-openfin/application/actions/handlerActionCreator.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const createFSA_1 = require("../../utils/createFSA");
const actionTypes = require("./actionTypes");
exports.startApplicationRes = createFSA_1.default(actionTypes.START_APPLICATION_RES, (options) => options);
exports.getCurrentRes = createFSA_1.default(actionTypes.APPLICATION_GET_CURRENT_RES, (options) => options);
exports.getWindowRes = createFSA_1.default(actionTypes.APPLICATION_GET_WINDOW_RES, (options) => options);
exports.wrapRes = createFSA_1.default(actionTypes.APPLICATION_WRAP_RES, (options) => options);
exports.addListenerRes = createFSA_1.default(actionTypes.APPLICATION_ADD_LISTENER_RES, (options) => options);
exports.quitRes = createFSA_1.default(actionTypes.APPLICATION_QUIT_RES, (options) => options);
exports.getChildWindowsRes = createFSA_1.default(actionTypes.APPLICATION_GET_CHILD_WINDOWS_RES, (options) => options);
exports.getInfoRes = createFSA_1.default(actionTypes.APPLICATION_GET_INFO_RES, (options) => options);
exports.getShortcutsRes = createFSA_1.default(actionTypes.APPLICATION_GET_SHORTCUTS_RES, (options) => options);
exports.getTrayIconInfoRes = createFSA_1.default(actionTypes.APPLICATION_GET_TRAY_ICON_INFO_RES, (options) => options);
exports.getZoomLevelRes = createFSA_1.default(actionTypes.APPLICATION_GET_ZOOM_LEVEL_RES, (options) => options);
exports.isRunningRes = createFSA_1.default(actionTypes.APPLICATION_IS_RUNNING_RES, (options) => options);
exports.removeListenerRes = createFSA_1.default(actionTypes.APPLICATION_REMOVE_LISTENER_RES, (options) => options);
exports.restartRes = createFSA_1.default(actionTypes.RESTART_RES, (options) => options);
exports.scheduleRestartRes = createFSA_1.default(actionTypes.APPLICATION_SCHEDULE_RESTART_RES, (options) => options);
exports.setShortcutsRes = createFSA_1.default(actionTypes.APPLICATION_SET_SHORTCUTS_RES, (options) => options);
exports.setTrayIconRes = createFSA_1.default(actionTypes.APPLICATION_SET_TRAY_ICON_RES, (options) => options);
exports.setZoomLevelRes = createFSA_1.default(actionTypes.APPLICATION_SET_ZOOM_LEVEL_RES, (options) => options);
exports.terminateRes = createFSA_1.default(actionTypes.APPLICATION_TERMINATE_RES, (options) => options);
