// redux-openfin/index.js
/** @license Openfin browser adapter v1.0.0
     *
     * This source code is licensed under the MIT license found in the
     * LICENSE file in the root directory of this source tree.
     */
    "use strict";
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
__export(require("./GlobalTypes"));
var makeType_1 = require("./utils/makeType");
exports.isReduxOpenfinAct = makeType_1.isReduxOpenfinAct;
exports.isReqAct = makeType_1.isReqAct;
exports.isResAct = makeType_1.isResAct;
var middleware_1 = require("./middleware");
exports.createOpenfinMiddleware = middleware_1.middlewareCreator;
var application_1 = require("./application");
exports.Applcation = application_1.default;
var docking_1 = require("./docking");
exports.Docking = docking_1.default;
var event_1 = require("./event");
exports.Event = event_1.default;
var interApplicationBus_1 = require("./interApplicationBus");
exports.InterApplicationBus = interApplicationBus_1.default;
var notification_1 = require("./notification");
exports.Notification = notification_1.default;
var window_1 = require("./window");
exports.Window = window_1.default;
var system_1 = require("./system");
exports.System = system_1.default;
