// redux-openfin/init.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const channel_1 = require("./channel");
const registerDefaultListener_1 = require("./event/registerDefaultListener");
const init_1 = require("./docking/init");
var ChannelType;
(function (ChannelType) {
    ChannelType[ChannelType["STANDALONE"] = 1] = "STANDALONE";
    ChannelType[ChannelType["PROVIDER"] = 2] = "PROVIDER";
    ChannelType[ChannelType["CLIENT"] = 3] = "CLIENT";
})(ChannelType = exports.ChannelType || (exports.ChannelType = {}));
exports.initState = {
    fin: null,
    store: null,
    channel: null,
    config: null,
    currentWindow: null,
    currentApplication: null,
};
let initialized = false;
function init(fin, config, store) {
    if (!initialized) {
        channel_1.default(fin, config, store).catch(e => {
            throw e;
        });
        initialized = true;
        exports.initState.config = config;
        if (store) {
            registerDefaultListener_1.default(fin, store);
        }
        exports.initState.fin = fin;
        if (fin && fin) {
            exports.initState.currentApplication = fin.Application.getCurrentSync();
            exports.initState.currentWindow = fin.Window.getCurrentSync();
            if (config.autoDocking && config.channelType === ChannelType.PROVIDER) {
                const dockingOptions = config.dockingOptions ? config.dockingOptions : {};
                init_1.default(fin, exports.initState.currentWindow, dockingOptions);
            }
        }
        if (config.ignoreStore) {
            exports.initState.store = null;
        }
        else {
            exports.initState.store = store;
        }
    }
}
exports.default = init;
