// redux-openfin/window/actions/actionCreators.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const createFSA_1 = require("../../utils/createFSA");
const actionTypes = require("./actionTypes");
exports.getCurrent = createFSA_1.default(actionTypes.GET_CURRENT, (options) => options);
exports.wrap = createFSA_1.default(actionTypes.WRAP, (options) => options);
exports.addListener = createFSA_1.default(actionTypes.ADD_LISTENER, (options) => options);
exports.authenticate = createFSA_1.default(actionTypes.AUTHENTICATE, (options) => options);
exports.bringToFront = createFSA_1.default(actionTypes.BRING_TO_FRONT, (options) => options);
exports.createWindow = createFSA_1.default(actionTypes.CREATE_WINDOW, (options) => options);
exports.close = createFSA_1.default(actionTypes.CLOSE, (options) => options);
exports.disableUserMovement = createFSA_1.default(actionTypes.DISABLE_USER_MOVEMENT, (options) => options);
exports.enableUserMovement = createFSA_1.default(actionTypes.ENABLE_USER_MOVEMENT, (options) => options);
exports.focus = createFSA_1.default(actionTypes.FOCUS, (options) => options);
exports.getBounds = createFSA_1.default(actionTypes.GET_BOUNDS, (options) => options);
exports.getGroup = createFSA_1.default(actionTypes.GET_GROUP, (options) => options);
exports.getState = createFSA_1.default(actionTypes.GET_STATE, (options) => options);
exports.getOptions = createFSA_1.default(actionTypes.GET_OPTIONS, (options) => options);
exports.hide = createFSA_1.default(actionTypes.HIDE, (options) => options);
exports.joinGroup = createFSA_1.default(actionTypes.JOIN_GROUP, (options) => options);
exports.leaveGroup = createFSA_1.default(actionTypes.LEAVE_GROUP, (options) => options);
exports.minimize = createFSA_1.default(actionTypes.MINIMIZE, (options) => options);
exports.maximize = createFSA_1.default(actionTypes.MAXIMIZE, (options) => options);
exports.mergeGroups = createFSA_1.default(actionTypes.MERGE_GROUPS, (options) => options);
exports.moveBy = createFSA_1.default(actionTypes.MOVE_BY, (options) => options);
exports.moveTo = createFSA_1.default(actionTypes.MOVE_TO, (options) => options);
exports.restore = createFSA_1.default(actionTypes.RESTORE, (options) => options);
exports.show = createFSA_1.default(actionTypes.SHOW, (options) => options);
exports.setAsForeground = createFSA_1.default(actionTypes.SET_AS_FOREGROUND, (options) => options);
exports.setBounds = createFSA_1.default(actionTypes.SET_BOUNDS, (options) => options);
exports.updateOptions = createFSA_1.default(actionTypes.UPDATE_OPTIONS, (options) => options);
