// redux-openfin/window/actions/actionTypes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const makeType_1 = require("../../utils/makeType");
exports.GET_CURRENT = makeType_1.makeReqType('GET_CURRENT');
exports.GET_CURRENT_RES = makeType_1.makeResType('GET_CURRENT_RES');
exports.WRAP = makeType_1.makeReqType('WRAP');
exports.WRAP_RES = makeType_1.makeResType('WRAP_RES');
exports.ADD_LISTENER = makeType_1.makeReqType('ADD_EVENT_LISTENER');
exports.ADD_LISTENER_RES = makeType_1.makeResType('ADD_EVENT_LISTENER_RES');
exports.AUTHENTICATE = makeType_1.makeReqType('AUTHENTICATE');
exports.AUTHENTICATE_RES = makeType_1.makeResType('AUTHENTICATE_RES');
exports.BRING_TO_FRONT = makeType_1.makeReqType('BRING_TO_FRONT');
exports.BRING_TO_FRONT_RES = makeType_1.makeResType('BRING_TO_FRONT_RES');
exports.CREATE_WINDOW = makeType_1.makeReqType('CREATE_WINDOW');
exports.CREATE_WINDOW_RES = makeType_1.makeResType('CREATE_WINDOW_RES');
exports.CLOSE = makeType_1.makeReqType('CLOSE');
exports.CLOSE_RES = makeType_1.makeResType('CLOSE_RES');
exports.DISABLE_USER_MOVEMENT = makeType_1.makeReqType('DISABLE_USER_MOVEMENT');
exports.DISABLE_USER_MOVEMENT_RES = makeType_1.makeResType('DISABLE_USER_MOVEMENT_RES');
exports.ENABLE_USER_MOVEMENT = makeType_1.makeReqType('ENABLE_USER_MOVEMENT');
exports.ENABLE_USER_MOVEMENT_RES = makeType_1.makeResType('ENABLE_USER_MOVEMENT_RES');
exports.FOCUS = makeType_1.makeReqType('FOCUS');
exports.FOCUS_RES = makeType_1.makeResType('FOCUS_RES');
exports.GET_GROUP = makeType_1.makeReqType('GET_GROUP');
exports.GET_GROUP_RES = makeType_1.makeResType('GET_GROUP_RES');
exports.GET_BOUNDS = makeType_1.makeReqType('GET_BOUNDS');
exports.GET_BOUNDS_RES = makeType_1.makeResType('GET_BOUNDS_RES');
exports.GET_STATE = makeType_1.makeReqType('GET_STATE');
exports.GET_STATE_RES = makeType_1.makeResType('GET_STATE_RES');
exports.GET_OPTIONS = makeType_1.makeReqType('GET_OPTIONS');
exports.GET_OPTIONS_RES = makeType_1.makeResType('GET_OPTIONS_RES');
exports.HIDE = makeType_1.makeReqType('HIDE');
exports.HIDE_RES = makeType_1.makeResType('HIDE_RES');
exports.JOIN_GROUP = makeType_1.makeReqType('JOIN_GROUP');
exports.JOIN_GROUP_RES = makeType_1.makeResType('JOIN_GROUP_RES');
exports.LEAVE_GROUP = makeType_1.makeReqType('LEAVE_GROUP');
exports.LEAVE_GROUP_RES = makeType_1.makeResType('LEAVE_GROUP_RES');
exports.MINIMIZE = makeType_1.makeReqType('MINIMIZE');
exports.MINIMIZE_RES = makeType_1.makeResType('MINIMIZE_RES');
exports.MAXIMIZE = makeType_1.makeReqType('MAXIMIZE');
exports.MAXIMIZE_RES = makeType_1.makeResType('MAXIMIZE_RES');
exports.MERGE_GROUPS = makeType_1.makeReqType('MERGE_GROUPS');
exports.MERGE_GROUPS_RES = makeType_1.makeResType('MERGE_GROUPS_RES');
exports.MOVE_BY = makeType_1.makeReqType('MOVE_BY');
exports.MOVE_BY_RES = makeType_1.makeResType('MOVE_BY_RES');
exports.MOVE_TO = makeType_1.makeReqType('MOVE_TO');
exports.MOVE_TO_RES = makeType_1.makeResType('MOVE_TO_RES');
exports.RESTORE = makeType_1.makeReqType('RESTORE');
exports.RESTORE_RES = makeType_1.makeResType('RESTORE_RES');
exports.SHOW = makeType_1.makeReqType('SHOW');
exports.SHOW_RES = makeType_1.makeResType('SHOW_RES');
exports.SET_AS_FOREGROUND = makeType_1.makeReqType('SET_AS_FOREGROUND');
exports.SET_AS_FOREGROUND_RES = makeType_1.makeResType('SET_AS_FOREGROUND_RES');
exports.SET_BOUNDS = makeType_1.makeReqType('SET_BOUNDS');
exports.SET_BOUNDS_RES = makeType_1.makeResType('SET_BOUNDS_RES');
exports.UPDATE_OPTIONS = makeType_1.makeReqType('UPDATE_OPTIONS');
exports.UPDATE_OPTIONS_RES = makeType_1.makeResType('UPDATE_OPTIONS_RES');
