// redux-openfin/window/actions/handlerActionCreator.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const createFSA_1 = require("../../utils/createFSA");
const actionTypes = require("./actionTypes");
exports.getCurrentRes = createFSA_1.default(actionTypes.GET_CURRENT_RES, (options) => options);
exports.wrapRes = createFSA_1.default(actionTypes.WRAP_RES, (options) => options);
exports.addListenerRes = createFSA_1.default(actionTypes.ADD_LISTENER_RES, (options) => options);
exports.authenticateRes = createFSA_1.default(actionTypes.AUTHENTICATE_RES, (options) => options);
exports.bringToFrontRes = createFSA_1.default(actionTypes.BRING_TO_FRONT_RES, (options) => options);
exports.createWindowRes = createFSA_1.default(actionTypes.CREATE_WINDOW_RES, (options) => options);
exports.closeRes = createFSA_1.default(actionTypes.CLOSE_RES, (options) => options);
exports.disableFrameRes = createFSA_1.default(actionTypes.DISABLE_USER_MOVEMENT_RES, (options) => options);
exports.enableFrameRes = createFSA_1.default(actionTypes.ENABLE_USER_MOVEMENT_RES, (options) => options);
exports.focusRes = createFSA_1.default(actionTypes.FOCUS_RES, (options) => options);
exports.getBoundsRes = createFSA_1.default(actionTypes.GET_BOUNDS_RES, (options) => options);
exports.getGroupRes = createFSA_1.default(actionTypes.GET_GROUP_RES, (options) => options);
exports.getStateRes = createFSA_1.default(actionTypes.GET_STATE_RES, (options) => options);
exports.getOptionsRes = createFSA_1.default(actionTypes.GET_OPTIONS_RES, (options) => options);
exports.hideRes = createFSA_1.default(actionTypes.HIDE_RES, (options) => options);
exports.joinGroupRes = createFSA_1.default(actionTypes.JOIN_GROUP_RES, (options) => options);
exports.leaveGroupRes = createFSA_1.default(actionTypes.LEAVE_GROUP_RES, (options) => options);
exports.minimizeRes = createFSA_1.default(actionTypes.MINIMIZE_RES, (options) => options);
exports.maximizeRes = createFSA_1.default(actionTypes.MAXIMIZE_RES, (options) => options);
exports.mergeGroupsRes = createFSA_1.default(actionTypes.MERGE_GROUPS_RES, (options) => options);
exports.moveByRes = createFSA_1.default(actionTypes.MOVE_BY_RES, (options) => options);
exports.moveToRes = createFSA_1.default(actionTypes.MOVE_TO_RES, (options) => options);
exports.restoreRes = createFSA_1.default(actionTypes.RESTORE_RES, (options) => options);
exports.showRes = createFSA_1.default(actionTypes.SHOW_RES, (options) => options);
exports.setAsForegroundRes = createFSA_1.default(actionTypes.SET_AS_FOREGROUND_RES, (options) => options);
exports.setBoundsRes = createFSA_1.default(actionTypes.SET_BOUNDS_RES, (options) => options);
exports.updateOptionsRes = createFSA_1.default(actionTypes.UPDATE_OPTIONS_RES, (options) => options);
